### ISSUE TITLE
[FEATURE] Unity table tag support in databricks_sql_table

### ISSUE THREAD

**Issue by donatasm** [databricks/terraform-provider-databricks#3514]
### Proposal

Similar to existing key value definitions in `databricks_sql_table` like [options](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/sql_table#options) or [properties](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/sql_table#properties) proposal is to have `tags` attribute:

```hcl
databricks_sql_table {
    name = "test_table"
    ...
    tags = {
        key1 = value1
        key2 = value2
    }
}
```

Implementation in [resource_sql_table.go](https://github.com/databricks/terraform-provider-databricks/blob/b1703a4ece7b8a6617ced16d60c51c9616af9154/catalog/resource_sql_table.go) would additionally execute:

```sql
ALTER TABLE 'test_table' SET tags ('key1' = 'value1', 'key2' = 'value2')
```

**Comment by coenvd** [comment]
Whats the current status on this? Its quite strange that we cannot do this with Terraform.

**Comment by vsluc** [comment]
I agree with @coenvd . Could we please push for adding the necessary API and terraform feature?
